File: sort-java/src/Shell.java
public class Shell {
    public Shell(){};
    public int[] sort(int[] a){
        int gap,len;
        len=a.length;
        for (gap=len/2;gap>0;gap=gap/2){
            for (int i=0;i<gap;i++){
                for (int j=i+gap;j<len;j+=gap){
                    if (a[j]<a[j-gap]){
                        int tmp=a[j];
                        int k=j-gap;
                        while (k>=0&&a[k]>tmp){
                            a[k+gap]=a[k];
                            k=k-gap;
                        }
                        a[k+gap]=tmp;
                    }
                }
            }
        }
        return a;
    }
}

File: sort-python/Shell.py
class shell:
    def shellsort(self,a):
        gap=len(a)//2
        while gap>0:
            for i in range(0,gap):
                for j in range(i+gap,len(a),gap):
                    if a[j]<a[j-gap]:
                        tmp=a[j]
                        k=j-gap
                        while k>=0 and a[k]>tmp:
                            a[k+gap]=a[k]
                            k=k-gap
                        a[k+gap]=tmp
            gap=gap//2
        return a
